RCAIDE/Framework/Networks/Electric.py

```diff
@@ -132,87 +142,124 @@ def evaluate(self,state,center_of_gravity):
                                     else:
                                         # use previous propulsor results 
                                         T,M,P = propulsor.reuse_stored_data(state,bus,stored_propulsor_tag,center_of_gravity)
-                                  
+
                                 total_thrust += T   
                                 total_moment += M   
                                 total_power  += P 
-                        
+
                         # compute power from each componemnt 
                         avionics_power  = (avionics_conditions.power*battery.bus_power_split_ratio)/len(batteries)* state.ones_row(1) 
                         payload_power   = (payload_conditions.power*battery.bus_power_split_ratio)/len(batteries) * state.ones_row(1)   
                         charging_power  = (state.conditions.energy[bus.tag][battery.tag].pack.charging_current*bus_voltage*battery.bus_power_split_ratio)/len(batteries) 
                         total_esc_power = total_power*battery.bus_power_split_ratio  
-                           
+
                         # append bus outputs to battery 
                         battery_conditions                    = state.conditions.energy[bus.tag][battery.tag] 
                         battery_conditions.pack.power_draw    = ((avionics_power + payload_power + total_esc_power) - charging_power)/bus.efficiency
                         battery_conditions.pack.current_draw  = battery_conditions.pack.power_draw/bus_voltage
-                        battery.energy_calc(state,bus,recharging_flag)        
-                
+                       
+        
+        time               = state.conditions.frames.inertial.time[:,0] 
+        delta_t            = np.diff(time)
+        for t_idx in range(state.numerics.number_of_control_points):
```

Comment:
I thought we loop over n-1 points here since the first timestep is already known. i.e. should t_idx  go from  0 to n-1?

Reply:
It does go through till n-1;
this is done by  
# ---------------------------------------------------------------------------------------------------     
 # Future State 
 # --------------------------------------------------------------------------------------------------- 
    if t_idx != state.numerics.number_of_control_points-1:  

        # Compute cell temperature